src/main/java/net/minestom/server/listener/TabCompleteListener.java

```diff
@@ -15,13 +16,27 @@
 import net.minestom.server.utils.StringUtils;
 
 import java.util.Arrays;
+import java.util.Optional;
 import java.util.regex.Pattern;
 
 public class TabCompleteListener {
 
     public static void listener(ClientTabCompletePacket packet, Player player) {
         final String text = packet.text();
+        getSuggestion(player, text)
+                .ifPresent(suggestion ->
+                        player.sendPacket(new TabCompletePacket(
+                                packet.transactionId(),
+                                suggestion.getStart(),
+                                suggestion.getLength(),
+                                suggestion.getEntries().stream()
+                                        .map(suggestionEntry -> new TabCompletePacket.Match(suggestionEntry.getEntry(), suggestionEntry.getTooltip()))
+                                        .toList())
+                        )
+                );
+    }
 
+    public static Optional<Suggestion> getSuggestion(CommandSender commandSender, String text) {
```

Comment:
I'd personally just return a nullable suggestion. Not much reason to deal with the overhead of `Optional` (+lambda) here since it is purely internal